wallet-api/src/WalletAPI/Internal/Crypto.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE GADTs #-}
+
+module WalletAPI.Internal.Crypto where
+
+import           Crypto.Cipher.AES (AES256)
+import           Crypto.Cipher.Types (BlockCipher(..), Cipher(..), nullIV, KeySizeSpecifier(..), IV, makeIV)
+import           Crypto.Error (CryptoFailable(..), CryptoError(..))
+
+import qualified Crypto.Random.Types as CRT
+
+import           Data.Functor
+import           Data.ByteArray (ByteArray)
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString.Base16  as Hex
+import           Data.Aeson
+import qualified Data.Text.Encoding      as T
+
+-- | Not required, but most general implementation
+data Key c a where
```

Comment:
Should we separate models and functions using different files?

Reply:
Let's keep data and syntax for it in the same file.